### PR TITLE
ci: Add doctests to CI again.

### PR DESCRIPTION
**Pull request by hameerabbasi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] 🪄 Feature
- [ ] 🐞 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📚 Documentation
- [x] 🧪 Test
- [ ] 🛠️ Other

## Related issues

- Related issue #
- Closes #

## Checklist

- [x] Code follows style guide
- [x] Tests added
- [x] Documented the changes

***

### CODE REVIEW

**Comment by hameerabbasi** [comment]
@willow-ahrens Doctest issue.

**Comment by willow-ahrens** [comment]
using numba from the numba channel (with a statically linked llvmlite) fixes the segfault issue on mac. It appears that the segfault issue is mac-only at this point, as linux supports some special symbol resolution thing that fixed this problem.

Proposed solution: I would like to leave a warning to users on startup in the event that they have tried to load this particular config of packages. I think we could warn in pydata/sparse whenever we see (conda in use) AND (macos) AND (numba was not installed from numba channel)

As a side note: it is odd to me that we are loading both numba and finch at the same time. we might reduce some of this headache by switching between backends as a build option, rather than a runtime option.

**Comment by willow-ahrens** [comment]
I cannot get a reproducer for this in docker, though I have tried. If anyone can get a reproducer for this that would be great, then we can put it in an issue where we describe the problem. If you ssh into the CI runner, even doing "import sparse.finch_backend" will segfault, so we don't need all the pytest stuff to reproduce.

**Comment by willow-ahrens** [comment]
I've put a comment in to pyjuliapackage that we experience an issue with parallel installs, and suggested a fix here: https://github.com/JuliaPy/pyjuliapkg/issues/19

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -126,6 +131,9 @@ jobs:
         SPARSE_BACKEND: ${{ matrix.backend }}
       run: |
         cd ${GITHUB_WORKSPACE}/array-api-tests
+        if [ "${SPARSE_BACKEND}" = "Finch" ]; then
+            python -c 'import finch'
+        fi
```
Comment (mtsokol) [inline .github/workflows/ci.yml:136]: Why do we need these lines?

**Comment by willow-ahrens** [comment]
also, note this comment regarding installation of juliacall and numba on conda: https://github.com/JuliaPy/PythonCall.jl/issues/215#issuecomment-2540274772

**Review by mtsokol — APPROVED** [review]
LGTM! Just one question.

**Comment by mtsokol** [comment]
And please remember about a commit squash!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (willow-ahrens) [reply .github/workflows/ci.yml:136]: thats the parallel julia install problem: https://github.com/JuliaPy/pyjuliapkg/issues/19

**Pull request merged**